plugins/modules/monitoring/pagerduty_user.py

```diff
@@ -0,0 +1,266 @@
+
+    # add a user to a team/teams
+    def add_user_to_teams(self, pd_user_id, pd_team, pd_role):
+        updated_team = None
+        for team in pd_team:
+            team_info = self._apisession.find('teams', team, attribute='name')
+            if team_info is not None:
+                try:
+                    updated_team = self._apisession.rput('/teams/'+team_info['id']+'/users/'+pd_user_id, json={
+                        'role':pd_role
+                    })
+                except PDClientError:
+                    updated_team = None
+        return updated_team
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            access_token=dict(type='str', required=True, no_log=True),
+            pd_user=dict(type='str', required=True),
+            pd_email=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            pd_role=dict(type='str', default='responder', required=False, choices=['global admin', 'manager', 'responder', 'observer', 'stakeholder', 'limited skateholder', 'restricted access']),
+            pd_team=dict(type='list', required=False)
+        ),
+        required_if=[
+            ['state', 'present', ['pd_team']],
+        ],
+        supports_check_mode=True,
+    )
+
+    if not HAS_PD_PY:
+        module.fail_json(msg=missing_required_lib('pdpyras', url='https://github.com/PagerDuty/pdpyras'), exception=PD_IMPORT_ERR)
+
+    if not HAS_PD_CLIENT_ERR:
+        module.fail_json(msg=missing_required_lib('PDClientError', url='https://github.com/PagerDuty/pdpyras'), exception=PD_CLIENT_ERR_IMPORT_ERR)
+
+    access_token = module.params['access_token']
+    pd_user = module.params['pd_user']
+    pd_email = module.params['pd_email']
+    state = module.params['state']
+    pd_role = module.params['pd_role']
+    pd_team = module.params['pd_team']
+
+    if pd_role:
+        pd_role_gui_value = {
+            'global admin': 'admin',
+            'manager': 'user',
+            'responder': 'limited_user',
+            'observer': 'observer',
+            'stakeholder': 'read_only_user',
+            'limited stakeholder': 'read_only_limited_user',
+            'restricted access': 'restricted_access'
+        }
+        pd_role = pd_role_gui_value[pd_role]
+
+    # authenticate with PD API
+    try:
+        session = APISession(access_token)
+    except PDClientError as e:
+        module.fail_json(msg="Failed to authenticate with PagerDuty: %s" % e)
+
+    user = PagerDutyUser(module, session)
+
+    user_exists = user.does_user_exist(pd_email)
+
+    if user_exists:
+        if state == "absent":
+            # remove user
+            if not module.check_mode:
+                user.delete_user(user_exists, pd_user)
+            module.exit_json(changed=True, result="Successfully deleted user %s" % pd_user)
+        else:
+            module.exit_json(changed=False, result="User %s already exists." % pd_user)
```

Comment:
What if some of the properties changed, like `pd_user` (since you search by email), `pd_role` and/or `pd_team`? Users often expect Ansible modules to ensure that the data is set to the current state, i.e. to check whether the data is as provided in the options, and it if is not, (try) to change the data.

Reply:
@felixfontein, users are deleted via user id (could be something like this: PXPGF42) which is returned through does_user_exist and passed to delete method/function.